tests/windowflags.py

```diff
@@ -0,0 +1,189 @@
+
+windowFlags1 = QtCore.Qt.WindowFlags()
+windowFlags2 = QtCore.Qt.WindowFlags()
+
+windowFlagsTest = windowFlags1     # type: QtCore.Qt.WindowFlags
+
+
+# window flags may be created by combining windowFlags together
+windowFlagsTest = ~windowFlags1
+windowFlagsTest = windowFlags1 | windowFlags2
+windowFlagsTest = windowFlags1 & windowFlags2
+windowFlagsTest = windowFlags1 ^ windowFlags2
+
+# window flags may be created by combining windowType together
+windowFlagsTest = ~windowType1
+windowFlagsTest = windowType1 & windowType2
+windowFlagsTest = windowType1 | windowType2
+windowFlagsTest = windowType1 ^ windowType2
+windowFlagsTest = windowType1 + windowType2
+windowFlagsTest = windowType1 - windowType2
+
+# window flags may also be created by combining windowType and int, left or right
+windowFlagsTest = windowType1 & 33
+windowFlagsTest = windowType1 | 33
+windowFlagsTest = windowType1 ^ 33
+windowFlagsTest = windowType1 + 33
+windowFlagsTest = windowType1 - 33
+windowFlagsTest =  33 & windowType1
+windowFlagsTest =  33 | windowType1
+windowFlagsTest =  33 ^ windowType1
+windowFlagsTest =  33 + windowType1
+windowFlagsTest =  33 - windowType1
+
+# window flags may be created by combining windowFlags and windowType, left or right
+windowFlagsTest = windowFlags1 | windowType1
+windowFlagsTest = windowFlags1 & windowType1
+windowFlagsTest = windowFlags1 ^ windowType1
+windowFlagsTest = windowType1 | windowFlags1
+windowFlagsTest = windowType1 & windowFlags1
+windowFlagsTest = windowType1 ^ windowFlags1
+
+
+# window flags may be created by combining windowFlags and int, right only
+windowFlagsTest = windowFlags1 | 33
+windowFlagsTest = windowFlags1 & 33
+windowFlagsTest = windowFlags1 ^ 33
+
+
+# this is rejected and is slightly annoying: you can not pass a WindowType variable to a method expecting a WindowFlags
+# explicit typing must be used on those methods to accept both WindowType and WindowFlags
+windowFlagsTest = windowType1   # type: ignore
+
+# correct way to do it
+windowFlagsTest = QtCore.Qt.WindowFlags(windowType1)
+
+# this is rejected for the same reason as for windowType. We want windowFlags to be stay precise typing
+intValue = windowFlagsTest      # type: ignore
+
+# correct way to do it
+intValue = int(windowFlagsTest)
+
+# rejected because all other operations on WindowFlags would then fail, because int
+# does not support as many operations as WindowFlags
+windowFlagsTest = 33            # type: ignore
+
+# correct way to do it
+windowFlagsTest = QtCore.Qt.WindowFlags(33)
+
+
+#########################################################3
+#
+#        Exploring errors
+#
+#########################################################3
+
+# This checks the following:
+# + and - operations are not supported on windowFlags
+# combining int with windowFlags does not work
+
+try:
+    windowFlagsTest = windowFlags1 + windowFlags2		# type: ignore
+    assert False
+except TypeError:
+    pass
```

Comment:
I ran it without py.test because it would not work very well on my computer. I am not sure how much you have tested all the flow on Windows but I was not interested in debugging it. Anyway, none of the other tests rely on py.test so this was as good as it gets.

Reply:
Thanks for the explanation.  I've only done https://github.com/python-qt-tools/PyQt5-stubs/pull/123 for Windows so far...  `:[`